parser/parser_test.go

```diff
@@ -29,42 +29,96 @@ func TestParserUtils(t *testing.T) {
 		require.Nil(t, p.err)
 	})
 
-	t.Run("test many can read array", func(t *testing.T) {
-		p := newParser("[a b c d]")
+	t.Run("test many", func(t *testing.T) {
+		t.Run("can read array", func(t *testing.T) {
+			p := newParser("[a b c d]")
+
+			var arr []string
+			p.many(lexer.BracketL, lexer.BracketR, func() {
+				arr = append(arr, p.next().Value)
+			})
+			require.Nil(t, p.err)
+			require.Equal(t, []string{"a", "b", "c", "d"}, arr)
+
+			require.Equal(t, lexer.EOF, p.peek().Kind)
+			require.Nil(t, p.err)
+		})
+
+		t.Run("return if open is not found", func(t *testing.T) {
+			p := newParser("turtles are happy")
 
-		var arr []string
-		p.many(lexer.BracketL, lexer.BracketR, func() {
-			arr = append(arr, p.next().Value)
+			p.many(lexer.BracketL, lexer.BracketR, func() {
+				t.Error("cb should not be called")
+			})
+			require.Nil(t, p.err)
+			require.Equal(t, "turtles", p.next().Value)
 		})
-		require.Nil(t, p.err)
-		require.Equal(t, []string{"a", "b", "c", "d"}, arr)
 
-		require.Equal(t, lexer.EOF, p.peek().Kind)
-		require.Nil(t, p.err)
+		t.Run("will stop on error", func(t *testing.T) {
+			p := newParser("[a b c d]")
+
+			var arr []string
+			p.many(lexer.BracketL, lexer.BracketR, func() {
+				arr = append(arr, p.next().Value)
+				if len(arr) == 2 {
+					p.error(p.peek(), "boom")
+				}
+			})
+			require.EqualError(t, p.err, "input.graphql:1: boom")
+			require.Equal(t, []string{"a", "b"}, arr)
+		})
 	})
 
-	t.Run("test many return if open is not found", func(t *testing.T) {
-		p := newParser("turtles are happy")
+	t.Run("test some", func(t *testing.T) {
+		t.Run("can read array", func(t *testing.T) {
+			p := newParser("[a b c d]")
+
+			var arr []string
+			p.some(lexer.BracketL, lexer.BracketR, func() {
+				arr = append(arr, p.next().Value)
+			})
+			require.Nil(t, p.err)
+			require.Equal(t, []string{"a", "b", "c", "d"}, arr)
 
-		p.many(lexer.BracketL, lexer.BracketR, func() {
-			t.Error("cb should not be called")
+			require.Equal(t, lexer.EOF, p.peek().Kind)
+			require.Nil(t, p.err)
 		})
-		require.Nil(t, p.err)
-		require.Equal(t, "turtles", p.next().Value)
-	})
 
-	t.Run("test many will stop on error", func(t *testing.T) {
-		p := newParser("[a b c d]")
+		t.Run("can't read empty array", func(t *testing.T) {
+			p := newParser("[]")
+
+			var arr []string
+			p.some(lexer.BracketL, lexer.BracketR, func() {
+				arr = append(arr, p.next().Value)
+			})
+			require.EqualError(t, p.err, "input.graphql:1: expected at least one definition, found ]")
+			require.Equal(t, []string(nil), arr)
+			require.NotEqual(t, lexer.EOF, p.peek().Kind)
+		})
+
+		t.Run("return if open is not found", func(t *testing.T) {
+			p := newParser("turtles are happy")
+
+			p.some(lexer.BracketL, lexer.BracketR, func() {
+				t.Error("cb should not be called")
+			})
+			require.Nil(t, p.err)
```

Comment:
should this be `expected [, found turtles`?

Reply:
`some` method has below code.
```
	hasDef := p.skip(start)
	if !hasDef {
		return
	}
```

We should consume `type Foo @bar` but `type Foo {}` is not.

Reply:
name of method is incorrect? 🤔